### DISCUSSION TITLE
How to update a Blitz.js application?

### DISCUSSION THREAD

**Discussion by fnn** [blitz-js/blitz#992] [Q&A] [answered]
There are a lot of new releases. Is there a way to update an existing app with all the new improvements in the generator template or do I have to do it manually?

**Comment by flybayer** [comment, marked as answer]
Good question!

Probably the simplest way is to do this:

1. `npm i -g blitz@latest`
2. Go to the folder containing your app. If app is at `~/projects/myApp`, then `cd ~/projects/`
3. `blitz new myApp`

Then you'll it'll say there are file conflicts and you'll be able to see the diff of each file.

We could add some type of command specifically for this case too.

**Reply by fnn** [reply]
Thanks, have not thought of `blitz new`.

**Reply by nabeelkausari** [reply]
Thanks, this was extremely helpful

**Reply by tristen** [reply]
If I follow the instructions above, an existing application is not updated. Instead, a new folder of the same name and scaffold is created. I'm also seeing `(global)` detected as the `(local)` version.

Running `blitz --version` returns:

```
blitz: 0.36.3 (global)
blitz: 0.35.0 (local)
```

Running `blitz new {appname}` within the folder containing my app returns:

```
? Your global blitz version is outdated. Upgrade? …
❯ Yes - Upgrade to 0.36.3
  No - Continue with old version (0.35.0) - NOT recommended
```

> We could add some type of command specifically for this case too.

+1 Something like `blitz update` would be nice!

**Reply by flybayer** [reply]
Let me clarify:

1. For an app at `~/projects/myApp`:
2. `cd ~/projects`
3. `blitz new myApp`

Note that you don't run blitz new from `~/projects/myApp/`.

Related: there's an issue with the yarn auto updating: https://github.com/blitz-js/blitz/issues/2415

**Reply by tristen** [reply]
Ah! appreciate the clarification 🤦

**Reply by Vandivier** [reply]
@flybayer is this still up to date? it's a top google result for `blitz js upgrade`

**Reply by flybayer** [reply]
@Vandivier yes

**Reply by reymon359** [reply]
Hey @flybayer, is this still up to date?

Just asking because it's been more than a year and maybe there is a better way now 😄

**Reply by reymon359** [reply]
Hey @flybayer any update if this is the best way to upgrade blitz curently? 😄

**Reply by ourmaninamsterdam** [reply]
We're about to embark on an upgrade from an older v2 beta so would love some points on upgrade steps @flybayer
